### ISSUE TITLE
How to not display original photo (4.5 MB) but download and show only medium size.?

### ISSUE THREAD

**Issue by podporujeme** [buddypress/bp-attachments#104]
How to not display original photo (4.5 MB) but download and show only medium size.?
by some php snippet, or u add to plugin pls?

![imp](https://github.com/buddypress/bp-attachments/assets/141672673/65ecce7c-ad35-4eb9-ba0b-961fc3bf3bc3)

**Comment by imath** [comment]
Hi, the only way to do it so far is to privately upload the media. We surely need to offer a way to disable large image display: a possible way is to shrink resolution like WordPress does when generating image with specific sizes (thumbnail, medium, large. etc.

Thanks for sharing this point.

**Comment by podporujeme** [comment]
@imath for this plugin u can automatic resize image for max resolution 800px width... buddypress is "social network" People dont need view large photos.

**Comment by dcavins** [comment]
Ha, we can never assume we know how a plugin is going to be used. Users will always find a novel way to use something that surprises us as devs (and sometimes delights us, too). :D

I promise that some users will want to use it as a photo sharing repository and will want to store high-res images.

**Comment by podporujeme** [comment]
something like this? (if is blank or zero is disable)
![Untitled](https://github.com/buddypress/bp-attachments/assets/141672673/b6672fb4-83ae-4e14-b7bb-4641ef0cb57c)

**Comment by imath** [comment]
I agree with @dcavins I believe the best way is to generate a shrinked image for preview and the original one for download.

**Comment by podporujeme** [comment]
Yes that would be great.

**Comment by bclaim** [comment]
@imath Just putting in my vote for this too. If we're able to control this through constants, like in BuddyPress, that would be awesome!

An example could be:

> define ( 'BP_ATTACHMENTS_THUMB_WIDTH', 350 );
> define ( 'BP_ATTACHMENTS_THUMB_HEIGHT', 400 );
> define ( 'BP_ATTACHMENTS_FULL_WIDTH', 700 );
> define ( 'BP_ATTACHMENTS_FULL_HEIGHT', 800 );
> define ( 'BP_ATTACHMENTS_ORIGINAL_MAX_WIDTH', 1024 );
> define ('BP_ATTACHMENTS_ORIGINAL_MAX_FILESIZE', 1024 );

**Comment by imath** [comment]
Hi @bclaim thanks for your suggestion 😍

**Comment by podporujeme** [comment]
@bclaim Hello, if i input this code to php snippet, it will work? Or i need some modification?
